tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -238,16 +291,61 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     return Status::OK();
   }
 
+  Status HandleMaximum(HloInstruction *instr) override {
+    HloInstruction *existing_gemm, *slice;
+    auto zero_pattern = m::Broadcast(m::ConstantScalar(0));
+    // Attempt to elide maximum and fuse ReLU activation into GEMM.
+    if (Match(instr, m::MaximumAnyOrder(zero_pattern,
+                                        m::Op(&existing_gemm)
+                                            .WithCustomCallTarget(
+                                                kCublasLtMatmulCallTarget)))) {
+      TF_ASSIGN_OR_RETURN(bool was_fused,
+                          FuseReluActivationSliced(instr, existing_gemm));
+      if (was_fused) {
+        return OkStatus();
+      }
+    }
+    // Attempt to elide maximum and bitcast and fuse ReLU activation into
+    // *batched* GEMM.
+    if (Match(instr, m::MaximumAnyOrder(
+                         zero_pattern,
+                         m::Bitcast(m::Op(&existing_gemm)
+                                        .WithCustomCallTarget(
+                                            kCublasLtMatmulCallTarget))))) {
+      TF_ASSIGN_OR_RETURN(bool was_fused,
+                          FuseReluActivationSliced(instr, existing_gemm));
+      if (was_fused) {
+        return OkStatus();
+      }
+    }
+    // Attempt to elide maximum and fuse ReLU activation into GEMM when slicing
+    // is applied to the result.
+    if (Match(instr, m::MaximumAnyOrder(
+                         m::Slice(&slice, m::Op(&existing_gemm)
+                                              .WithCustomCallTarget(
+                                                  kCublasLtMatmulCallTarget)),
+                         zero_pattern))) {
+      TF_ASSIGN_OR_RETURN(bool was_fused, FuseReluActivationSliced(
+                                              instr, existing_gemm, slice));
+      if (was_fused) {
+        return OkStatus();
+      }
+    }
+
+    return OkStatus();
+  }
+
   Status HandleConvert(HloInstruction *instr) override {
     HloInstruction *bias, *existing_gemm;
-    if (Match(
-            instr,
-            m::Convert(m::AddAnyOrder(
-                           m::Convert(m::Op(&existing_gemm)
-                                          .WithCustomCallTarget(kGemmCallTarget)
-                                          .WithElementType(BF16)),
-                           m::Convert(m::Op(&bias).WithElementType(BF16))))
-                .WithElementType(BF16))) {
+    if (Match(instr,
+              m::Convert(m::AddAnyOrder(
+                             m::Convert(m::Op(&existing_gemm)
+                                            .WithCustomCallTarget(
+                                                {kGemmCallTarget,
+                                                 kCublasLtMatmulCallTarget})
+                                            .WithElementType(BF16)),
+                             m::Convert(m::Op(&bias).WithElementType(BF16))))
+                  .WithElementType(BF16))) {
```

Comment:
This misses other idioms that have `convert` ops in the way, right?  Like, we handle bias add, but not e.g. relu?

Presumably we're also missing tests for this.

Reply:
I'm preparing a separate PR for Epilogue Fusion in BF16 which includes tests. This only shows as a change because I made a small correction.